### DISCUSSION TITLE
HOWTO / Cookbook / FAQ for qpdf

### DISCUSSION THREAD

**Discussion by m-holger** [qpdf/qpdf#1245] [Ideas]
In #1243, @rubyFeedback suggested:

> ... I was thinking, what if qpdf could have a FAQ that could showcase such small use cases? Really simple todo-lists, e. g. "I would like to do x, can you show me how to do so". Like a FAQ. Examples could also suffice, but most of the examples that seem to go with qpdf seem to be fairly small-ish. I'd ideally love to have a ton of examples, ideally in a single .md file, that showcase question-answer types, with a bit of explanation. Perhaps this already exists, and I can not find it, in which case it would be nice to make a link from the main README more prominently so. Perhaps it could be considered to improve the working-examples a bit, to showcase what qpdf can do, in particular when someone either does not know qpdf, or quickly forgets (I forget too quickly ... need to write this all down... which I forget to do so, too :P )

@mschilli87 commented:

> One place this could be added would be https://github.com/tldr-pages/tldr in case @jberkenbilt doesn't prefer to keep these in the repo (and thus take over the maintenance burden).

**Comment by m-holger** [comment]
I think in general the idea makes a lot of sense. The main problem I see, as has already been alluded to, is that time is short and the to-do list is long and growing. Realistically, the only way I see this taking off is if qpdf users rather than developers provide the content. The problem I foresee with qpdf users providing the content is that there is a chicken and egg situation - contributions would be much more likely if there was a good collection of 'how to' examples already available. Therefore, for this to take off I think volunteers are required to create some initial content.

In terms of the mechanics, one option would be to add a 'how to' section to the qpdf wiki. I am not sure about the suggested alternative of  https://github.com/tldr-pages/tldr . I had a very brief look and my initial feeling is that the format is a bit too brief - I think many of the tasks that would benefit from examples may require a little bit more explanation than the tldr format allows. That said, tldr would be better than nothing.

**Reply by jberkenbilt** [reply]
I agree with the above except I haven't actually looked at tldr. This has come up repeatedly over the years. I know that there are questions on Stack Overflow that have been answered with qpdf. I have contributed to not more than one or two of the answers, usually because someone has explicitly linked to a question there in an issue here. I have, at times, considered working answering Stack Overflow questions into my daily routine, but realistically this is not going to happen as I am already quite over-committed. With @m-holger as a second maintainer, there is more hope than there was before.

I think the best answer here would be for a knowledgeable community member to take the lead on this. If someone did this and tagged me to review solutions and potentially offer suggestions, I would happily contribute, but I can't sign up to start this or actively maintain it. If someone does this, I think it would be essential for people to always include the version of qpdf they had when they wrote the answer.
